Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs

```diff
@@ -20,42 +20,51 @@ namespace Microsoft.Dafny.LanguageServer.Language {
   /// this verifier serializes all invocations.
   /// </remarks>
   public class DafnyProgramVerifier : IProgramVerifier {
+    private readonly ILogger<DafnyProgramVerifier> logger;
     private readonly SemaphoreSlim mutex = new(1);
 
     public DafnyProgramVerifier(ILogger<DafnyProgramVerifier> logger) {
+      this.logger = logger;
     }
 
-    public async Task<IReadOnlyList<IImplementationTask>> GetVerificationTasksAsync(
-      ExecutionEngine engine,
+    public async Task<IReadOnlyList<IImplementationTask>> GetVerificationTasksAsync(ExecutionEngine engine,
       CompilationAfterResolution compilation,
+      ModuleDefinition moduleDefinition,
       CancellationToken cancellationToken) {
+
+      var verifiableModules = Translator.VerifiableModules(compilation.Program);
+      if (!verifiableModules.Contains(moduleDefinition)) {
+        throw new Exception();
+      }
+
       await mutex.WaitAsync(cancellationToken);
       try {
 
         var program = compilation.Program;
         var errorReporter = (DiagnosticErrorReporter)program.Reporter;
 
-        cancellationToken.ThrowIfCancellationRequested();
+        cancellationToken.ThrowIfCancellationRequested(); // TODO remove ?
```

Comment:
TODO? If you want to leave it, please put more explanations on what to do there.